### PR TITLE
fix: fix table outline frame calculations; removed conditional rendering of outline

### PR DESCRIPTION
**Pull request by jseibert**
The `TableExampleFactory` creates tables with outline borders, but the outlines are never drawn. This fixes that.

### CODE REVIEW

**Comment by philprime** [comment]
Hi Jeff, thanks for opening up this pull request.

I can confirm that the current example is not working as expected, and that your PR fixes it.

I do have a couple of comments, I'll add to the review

File: Source/Internal/Table/PDFTableObject.swift [hunk 1]
```diff
@@ -303,9 +304,12 @@ class PDFTableObject: PDFRenderObject {
                     cellFrame.origin.y += table.margin
                     contentFrame.origin.y -= startPosition.y - minOffset
                     contentFrame.origin.y += table.margin
-
-                    pageStart = pageStart == .null ? cellFrame.origin : pageStart
-                    pageEnd = CGPoint(x: cellFrame.maxX, y: cellFrame.maxY) + CGPoint(x: table.margin, y: table.margin)
+                    
+                    // Maintain pageStart and pageEnd
+                    pageStart.x = min(cellFrame.minX, pageStart.x)
```
Comment (philprime) [inline Source/Internal/Table/PDFTableObject.swift:309]: Do we have documentation available on how the min/max functions work with `CGPoint.null`?

File: Source/Internal/Table/PDFTableObject.swift [hunk 2]
```diff
@@ -395,24 +400,24 @@ class PDFTableObject: PDFRenderObject {
                                                     minOffset: minOffset,
                                                     maxOffset: maxOffset)
                 result += try sliceObject.calculate(generator: generator, container: container)
-
-                if nextPageCells.isEmpty && idx == cells.count - 1 {
```
Comment (philprime) [inline Source/Internal/Table/PDFTableObject.swift:399]: Unfortunately I didn't add comments 4 years ago when I added this conditional, maybe the logic has something to do with the multi-page breaking.

As the current implementation is not working, we'll go with this approach, and this comment is only for future reference

**Review by philprime — APPROVED** [review]

File: Source/Internal/Table/PDFTableObject.swift (hunk 1 above)
Comment (jseibert) [reply Source/Internal/Table/PDFTableObject.swift:309]: `.null` is defined in this codebase as `.infinity` so anything will be less. That's why I had to redefine `pageEnd` to `CGPoint()` aka `0,0` so `max` works as desired

**Comment by jseibert** [comment]
I'm not set up for iOS dev at the moment so I can't take a look at the failing tests, but I have thoroughly tested this on macOS and fixed one final bug with the final offsets.

**Comment by philprime** [comment]
Thank you, I can take over looking at the tests for iOS and fix it in the upcoming days

**Comment by philprime** [comment]
So I fixed the expected test results, they did not include the table outline to begin with.

But it seems like there is something wrong in the CI, going to fix this on a different branch, I'll then update this PR

**Comment by philprime** [comment]
alright looks like the checks are now passing.

Anymore changes from your side? Otherwise, lgtm

**Comment by jseibert** [comment]
Amazing thank you! It's all working for me, let's merge!

**Pull request merged**